ExampleCode/C++/API/API.cpp

```diff
@@ -0,0 +1,916 @@
+    // unless I made sure to make a string out of the upload_data picojson object,
+    // then with that string you can call c_str() on it below.
+    string upload_real = (value(upload_data).serialize());
+
+    // POST data. Upload will be the string with all the data.
+    curl_easy_setopt(curl, CURLOPT_POSTFIELDS, upload_real.c_str());
+
+    // JSON Headers
+    curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
+
+    // Verbose debug output - turn this on if you are having problems. It will spit out a ton
+    // of information, such as bytes sent off, headers/access/etc.
+    // Useful to see if you formatted the data right.
+    //curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
+
+    cout << "\nrSENSE response: \n";
+
+    // Perform the request, curl_code will get the return code
+    curl_code = curl_easy_perform(curl);
+
+    // This will put the HTTP response code into the "http_code" variable.
+    curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
+
+    /*
+     *                The iSENSE API gives us two response codes to check against:
+     *                Success: 200 OK (iSENSE handled the request fine)
+     *                Failure: 401 Unauthorized (email / password or contributor key was not valid)
+     *                Failure: 422 Unprocessable Entity (email or contributor key was fine,
+     *                         but there was an issue with the upload for some reason.)
+     *                the request's formatting. Something in the formatting caused iSENSE to fail.)
+     */
+
+    if(http_code == 200)
+    {
+      cout << "\n\nPOST request successfully sent off to iSENSE!\n";
+      cout << "The URL to your project is: " << dev_baseURL << "/projects/" << project_ID << endl;
+      return true;
+    }
+    else if(http_code == 401)
+    {
+      cout << "\nPOST request **failed**\n";
+      cout << "Try checking to make sure your contributor key is valid\n";
+      cout << "for the project you are trying to contribute to.\n";
+      return false;
+    }
+    else if(http_code == 422)
+    {
+      cout << "\n\nPOST request **failed**\n";
+      cout << "Something went wrong with iSENSE.\n";
+      cout << "Try formatting your data differently,\n";
+      cout << "using a contributor key instead of an email/password,\n";
+      cout << "or asking for help from others. You can also try running the\n";
+      cout << "the program with the \"debug\" method enabled, by typing: \n";
+      cout << "object_name.debug()\n";
+      cout << "This will output a ton of data to the console and may help you in\n";
+      cout << "debugging your program.\n";
+      return false;
+    }
+    else {
+      cout << "\n\nPOST request failed for some unknown reason. :/\n";
+      return false;
+    }
+
+    // For cURL return codes, see the following page:
+    // http://curl.haxx.se/libcurl/c/libcurl-errors.html
+    cout << "\ncURL return code was: " << curl_code << endl;

```

Comment:
Same `if/elseif/elseif/else` situation described earlier.  Doesn't look like the code below the else will be executed since every `if` block returns something.  Either that or maybe something isn't lining up right.